### PR TITLE
A few improvements

### PR DESCRIPTION
**Pull request by wtimme**
This branch improves the readability of the script a bit.

It also adds a `requirements.txt` file that allows for an easier management of dependencies (`pip install -r requirements.txt` is all that the user needs to execute).

Furthermore, the script now adds links to the issue keys in the changelog.

### CODE REVIEW

**Comment by k3muri84** [comment]
i will grab some of the suggestions, will check them all and let u know.
but not the `pip install -r requirements.txt` - because install `pip install jira` is much shorter and one file less to handle

File: README.md [hunk 1]
```diff
@@ -19,17 +19,27 @@ None of them is dealing with jira.
...
-- [jira-python](https://github.com/pycontribs/jira)
-    - install via `pip install jira`
+- Python 2.x
+  - `pip` for installing dependencies
 - jira admin rights - to create fix versions
 
-[1]: https://keepachangelog.com/en/1.0.0/
+## Setup
+
+On Python 3.3 and up, start the [virtual environment][1] in order to keep everything self-contained:
```
Comment (k3muri84) [inline README.md:28]: i think the script is not compatible with python3 yet, will add a todo to actually convert it to python3
also it's not required as setup here, more like a tip.

File: README.md [hunk 2]
```diff
@@ -19,17 +19,27 @@ None of them is dealing with jira.
 - [ ] optional: insert build types (e.g. beta or production)
 
 ## Requirements
-- python 2.x
-- [jira-python](https://github.com/pycontribs/jira)
-    - install via `pip install jira`
```
Comment (k3muri84) [inline README.md:24]: would actually keep that, it's easier than another extra file (with just one line)

File: README.md [hunk 3]
```diff
@@ -19,17 +19,27 @@ None of them is dealing with jira.
...
+    % pip install -r requirements.txt
+
+### Running the script
 
-## Usage
+Make sure your working directory is in the repository folder. Now, just execute it:
+
+    % ./generate-changelog.py
 
-### Quick Usage
```
Comment (k3muri84) [inline README.md:31]: I will add some more details to setup and usage, as it' is still a bit rough

File: generate-changelog.py [hunk 4]
```diff
@@ -11,13 +11,20 @@
 from jira import JIRA, JIRAError
 from datetime import datetime
 
-# point to your jira installation
-jira = JIRA(server=('https://jira.yourdomain.com'), auth=('changelogbot', 'cryp71cp455w0rd'))
+# The base URL of your Jira installation.
+# Must not have a trailing slash!
+jiraBaseUrl = 'https://jira.yourdomain.com'
```
Comment (k3muri84) [inline generate-changelog.py:16]: that's cool for the links in markdown - will grab this change

File: generate-changelog.py [hunk 5]
```diff
@@ -11,13 +11,20 @@
...
+
+jira = JIRA(server=(jiraBaseUrl), auth=('changelogbot', 'cryp71cp455w0rd'))
 # configure your jira project or just leave it to find all
 project_format = '[A-Z][A-Z]+'
 # define jira project to create version
 project_version = 'CORE'
 
+# Flag whether the script should automatically add the "version" information
+# to the issues in Jira.
+is_update_fix_versions_enabled = True
```
Comment (k3muri84) [inline generate-changelog.py:26]: for me thats the main feature :D will not take it, we can think about a dry run though if more people need that.

File: generate-changelog.py [hunk 6]
```diff
@@ -49,14 +56,18 @@ def load_properties(filepath, sep='=', comment_char='#'):
...
+
     fixVersions = []
     for existing_version in issue.fields.fixVersions:
         fixVersions.append({'name': existing_version.name})
     fixVersions.append({'name': version.name})
     try:
         issue.update(fields={'fixVersions': fixVersions})
     except JIRAError as e:
-        print e.status_code, e.text, issue.key
+        print(e.status_code, e.text, issue.key)
```
Comment (k3muri84) [inline generate-changelog.py:70]: prep for python3 ?

File: generate-changelog.py [hunk 7]
```diff
@@ -101,17 +112,22 @@ def scan_for_tickets():
     else:
         added.append(issue)
 
+def render_issue(issue):
```
Comment (k3muri84) [inline generate-changelog.py:115]: top

**Review by k3muri84 — COMMENTED** [review]
there is a lot of different changes here,
will grab the render function

**Comment by wtimme** [comment]
Hey @echorebel, thanks for taking a look! As you know, I've moved this logic to a fastlane action and will not pursue this Python script or the pull request for now.

**Pull request closed without merging**